From 15112323f9e74283cbb98b6e601418c25346c323 Mon Sep 17 00:00:00 2001
From: Cayetano Santos <csantosb@inventati.org>
Date: Thu, 15 Jan 2015 16:50:00 +0100
Subject: [PATCH] emacs: extend password-store-get to any field entry

now any field of an item may be retrieved, not only first line, in a
file formated similar to

password
item1: field1
item2: field2
etc.

when no item is provided, the functions behaves as before
---
 contrib/emacs/password-store.el | 17 ++++++++++++-----
 1 file changed, 12 insertions(+), 5 deletions(-)

diff --git a/contrib/emacs/password-store.el b/contrib/emacs/password-store.el
index 746f133..b3b2177 100644
--- a/contrib/emacs/password-store.el
+++ b/contrib/emacs/password-store.el
@@ -157,11 +157,18 @@ need to be commited manually if git is being used."
   (find-file (password-store--entry-to-file entry)))
 
 ;;;###autoload
-(defun password-store-get (entry)
-  "Return password for ENTRY.
-
-Returns the first line of the password data."
-  (car (s-lines (password-store--run-show entry))))
+(defun password-store-get (entry &optional item)
+  "Return field content of ITEM in ENTRY.
+
+When no ITEM is provided, returns the first line of ENTRY."
+  (if item
+      (let ((contents (s-lines (password-store--run-show entry))))
+	(catch 'loop
+	  (while contents
+	    (when (string-prefix-p item (car contents))
+	      (throw 'loop (s-trim-left (nth 1 (s-split item (car contents))))))
+	    (setq contents (cdr contents)))))
+    (car (s-lines (password-store--run-show entry)))))
 
 ;;;###autoload
 (defun password-store-clear ()
-- 
2.2.2

